### DISCUSSION TITLE
Cannot retrieve all the records using QueryDevicesByFilter()

### DISCUSSION THREAD

**Discussion by tputti2** [CrowdStrike/falconpy#240] [Q&A] [answered]
**Hi, I am trying to pull the device types using QueryDevicesbyFilter(), I can see that the total number of records is 11,100 but I am unable to pull more than 10,000 records** 

Code that prints the total number of records:
![image](https://user-images.githubusercontent.com/25337689/128214932-a312194f-27c4-479a-88b9-013496348c82.png)

**I am able to retrieve records from 0 to 10,000. I have to do 2 queries with each pulling 5000 records because I can only pull 5000 IDs in one call i.e. max value of 'limit' is 5000** 

![image](https://user-images.githubusercontent.com/74007258/128218113-b941af07-a3c0-48be-98e1-0dfa4d1c88d5.png)

![image](https://user-images.githubusercontent.com/74007258/128218508-d334a42c-f207-43b9-8ba8-19999554911d.png)

**I am unable to extract more than 10,000 records though I know the total number of records is greater than 10,000. Following is the code that gives an error. Can someone tell why this is happening? Thanks.**

![image](https://user-images.githubusercontent.com/25337689/128214022-cb4efc11-3e24-471a-8ad1-b67f13c104be.png)

**Comment by jshcodes** [comment, marked as answer]
Have you tried [QueryDevicesByFilterScroll](https://github.com/CrowdStrike/falconpy/wiki/Hosts#querydevicesbyfilterscroll)? I believe that's the method you should use when handling sets this large.
